Content.Shared/Item/ItemToggle/SharedItemToggleSystem.cs

```diff
@@ -0,0 +1,313 @@
+using Content.Shared.Interaction.Events;
+using Content.Shared.Toggleable;
+using Content.Shared.Weapons.Melee;
+using Content.Shared.Temperature;
+using Content.Shared.Wieldable;
+using Content.Shared.Wieldable.Components;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Network;
+
+namespace Content.Shared.Item.ItemToggle;
+/// <summary>
+/// Handles generic item toggles, like a welder turning on and off, or an e-sword.
+/// </summary>
+/// <remarks>
+/// If you need extended functionality (e.g. requiring power) then add a new component and use events.
+/// </remarks>
+public abstract class SharedItemToggleSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly SharedItemSystem _item = default!;
+    [Dependency] private readonly SharedPointLightSystem _light = default!;
+    [Dependency] private readonly INetManager _netManager = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ItemToggleComponent, UseInHandEvent>(OnUseInHand);
+        SubscribeLocalEvent<ItemToggleHotComponent, IsHotEvent>(OnIsHotEvent);
+        SubscribeLocalEvent<ItemToggleComponent, ItemUnwieldedEvent>(TurnOffonUnwielded);
+        SubscribeLocalEvent<ItemToggleComponent, ItemWieldedEvent>(TurnOnonWielded);
+        SubscribeLocalEvent<ItemToggleComponent, ItemToggleForceToggleEvent>(ForceToggle);
+    }
+    private void OnUseInHand(EntityUid uid, ItemToggleComponent itemToggle, UseInHandEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        args.Handled = true;
+
+        if (TryComp<WieldableComponent>(uid, out var wieldableComp))
+            return;
+
+        Toggle(uid, args.User, itemToggle);
+    }
+
+    public void Toggle(EntityUid uid, EntityUid? user = null, ItemToggleComponent? itemToggle = null)
+    {
+        if (!Resolve(uid, ref itemToggle))
+            return;
+
+        if (itemToggle.Activated)
+        {
+            TryDeactivate(uid, user, itemToggle: itemToggle);
+        }
+        else
+        {
+            TryActivate(uid, user, itemToggle: itemToggle);
+        }
+    }
+
+    public void ForceToggle(EntityUid uid, ItemToggleComponent itemToggle, ref ItemToggleForceToggleEvent args)
+    {
+        Toggle(uid, args.User, itemToggle);
+    }
```

Comment:
Unnecessary event as people can just call the method, also bit strange that this just calls toggle? It doesn't seem like it's forcing the state?

Reply:
Removed the method and event. Calling on the toggle method directly now.